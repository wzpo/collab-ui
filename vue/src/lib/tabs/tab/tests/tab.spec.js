import { mount } from '@vue/test-utils';
import Tab from '../index.vue';

describe('Tab', () => {
  it('should match snapshot', () => {
    const container = mount(Tab, {
      propsData: {
        heading: 'test'
      }
    });
    expect(container.html()).toMatchSnapshot();
  });

  it('should render one Tab', () => {
    const container = mount(Tab, {
      propsData: {
        heading: 'test'
      }
    });

    expect(container.findAll('li').length).toEqual(1);
  });

  it('should render TabHeader based on tabType prop', () => {
    const container = mount(Tab, {
      propsData: {
        heading: 'test'
      }
    });

    expect(container.find('md-tab-heading').exists()).toEqual(false);
  });

  it('should not be active by default', () => {
    const container = mount(Tab, {
      propsData: {
        heading: 'test'
      }
    });

    expect(container.find('li').classes('active')).toEqual(false);
  });

  it('should pass active prop', () => {
    const container = mount(Tab, {
      propsData: {
        heading: 'test',
        active: true
      }
    });

    expect(container.find('li').classes('active')).toEqual(true);
  });

  it('should handle onClick event', () => {
    let count = 0;
    const countUp = () => count++;
    const container = mount(Tab, {
      propsData: {
        heading: 'test'
      },
      listeners: {
        click: countUp
      }
    });

    container.find('a').trigger('click');
    expect(count).toEqual(1);
  });
});

