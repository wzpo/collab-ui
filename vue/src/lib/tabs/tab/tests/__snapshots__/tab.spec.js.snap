// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab should match snapshot 1`] = `<li tabindex="-1" class="md-tab__item"><a role="tab" href="javascript:void(0)" tabindex="-1">test</a></li>`;
